doc/conf.py

```diff
@@ -397,6 +400,7 @@ def reset_warnings(gallery_conf, fname):
     'show_memory': True,
     'line_numbers': False,  # XXX currently (0.3.dev0) messes with style
     'within_subsection_order': FileNameSortKey,
+    'filename_pattern': r'/.*(?<!_)\.py$',  # anything that isn't *_.py
```

Comment:
It looks like you got part of the way there but you'll need to rename `examples/datasets/spm_faces_dataset.py`, `examples/decoding/decoding_rsa.py`, `examples/visualization/make_report.py`, and four files under `examples/realtime/` to have trailing underscores